packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart

```diff
@@ -630,6 +651,56 @@ final class SignInStateMachine
     }
   }
 
+  /// Handles the MFA setup challenge.
+  @protected
+  Future<RespondToAuthChallengeRequest?> handleMfaSetup({
+    SignInEvent? event,
+    required bool hasUserResponse,
+  }) async {
+    final allowedMfaTypes = _allowedMfaTypes;
+    // Exclude MfaType.sms from consideration
+    final mfaTypesForSetup = allowedMfaTypes?.difference({MfaType.sms});
+    if (mfaTypesForSetup == null || mfaTypesForSetup.isEmpty) {
+      throw const InvalidUserPoolConfigurationException(
+        'No eligible MFA types are allowed for setup.',
+        recoverySuggestion: 'Check your user pool MFA configuration.',
+      );
+    }
+
+    if (mfaTypesForSetup.length == 1) {
+      final mfaType = mfaTypesForSetup.first;
+      if (mfaType == MfaType.totp) {
+        _enableMfaType = MfaType.totp;
+        _totpSetupResult ??= await associateSoftwareToken();
+        if (hasUserResponse) {
+          return createMfaSetupRequest(event as SignInRespondToChallenge);
+        } else {
+          // Need to prompt user for the TOTP code
+          return null;
```

Comment:
```suggestion
        return createMfaSetupRequest(event as SignInRespondToChallenge);
```
we do not need the if else condition if we call the handleMfaSetup when hasUserResponse